src/libs/actions/Report.js

```diff
@@ -756,6 +558,23 @@ function openReport(reportID, participantList = [], newReportObject = {}) {
         onyxData);
 }
 
+/**
+ * This will find an existing chat, or create a new one if none exists, for the given user or set of users. It will then navigate to this chat.
+ *
+ * @param {Array} userLogins list of user logins.
+ */
+function navigateToAndOpenReport(userLogins) {
+    const formattedUserLogins = _.map(userLogins, login => OptionsListUtils.addSMSDomainIfPhoneNumber(login).toLowerCase());
+    let newChat = {};
+    const chat = ReportUtils.getChatByParticipants(formattedUserLogins);
+    if (!chat) {
+        newChat = ReportUtils.buildOptimisticChatReport(formattedUserLogins);
+    }
```

Comment:
I don't have any great suggestions here, but it seems a bit confusing to have separate consts for chat|newChat when that distinction doesn't really matter

Reply:
I can add a comment if you think it's helpful, but the reason for that is if newChat is empty we do different things in openReport. So if `chat` were to have a value, we would still try to create the report in openReport

Reply:
Went ahead and added the comment. LMK if you think it needs to be more descriptive or anything :)

Reply:
thanks